number-46/index.html

```diff
@@ -0,0 +1,76 @@
+<!DOCTYPE html>
+<html>
+  <head>
+  </head>
+  <body>
+    <script>
+      class Shape {
+        constructor(length, width) {
+          this.length = length;
+          this.width = width;
+        }
+
+        get area() {
+          return this.length * this.width;
+        }
+
+        get name() {
+          return "Default Shape";
+        }
+      }
+
+      class Square extends Shape {
+        constructor(length) {
+          super(length, length);
+        }
+
+        get name() {
+          return "Square";
+        }
+      }
+
+      class Rectangle extends Shape {
+        constructor(length, width) {
+          super(length, width);
+        }
+
+        get name() {
+          return "Rectangle";
+        }
+      }
+
+      class Triangle extends Shape {
+        constructor(length, width) {
+          super(length, width / 2);
+        }
+
+        get name() {
+          return "Triangle";
+        }
+      }
+      class Circle extends Shape {
+        constructor(radius) {
+          super(radius * radius, Math.PI);
+        }
+
+        get name() {
+          return "Circle";
+        }
+      }
+      class Cube extends Shape {
+        constructor(length, width, height) {
+          super(length * width, height);
```

Comment:
This one might actually be better as something like this. You would also need to override the area function meaning you need to create a new one in this class.

```
constructor(length, width, height) {
  this.height = height;
  super(length, width);
}